endpoints/openrtb2/amp_auction.go

```diff
@@ -369,15 +370,15 @@ func (deps *endpointDeps) overrideWithParams(httpRequest *http.Request, req *ope
 	//In the AMP endpoint the consent string found in the http.Request query overrides that of the prebid query
 	queryConsentString := httpRequest.FormValue("gdpr_consent")
 	if queryConsentString != "" {
+		jsonMsg := json.RawMessage(`{"consent":"` + queryConsentString + `"}`)
 		// If nil, initialize
 		if req.User == nil {
-			req.User = &openrtb.User{}
+			req.User = &openrtb.User{Ext: jsonMsg}
+		} else { // else, keep whatever is in there and only substitute the consent string
+			re := regexp.MustCompile("^(.*consent\":\"?)[^,]*(\",.*)$")
+			repStr := "${1}" + queryConsentString + "$2"
+			req.User.Ext = json.RawMessage(re.ReplaceAllString(string(req.User.Ext), repStr))
```

Comment:
Won't this only work if the consent object already exists in the `ext`?

Reply:
You are right, adding an else clause there in case `req.User` is not null but `req.User.Ext` is.

Reply:
I don't think the replace will work if the `consent` object doesn't already exist.

Reply:
Corrected Hans